app/controllers/concerns/find_common.rb

```diff
@@ -1,12 +1,18 @@
-# this mixin is used by both ApplicationController and Api::BaseController
-# searches for an object based on its id, name, label, etc and assign it to an instance variable
+# This mixin is used by both ApplicationController and Api::BaseController
+# Searches for an object based on its id, name, label, etc and assign it to an instance variable
 # friendly_id performs the logic if params[:id] is 'id' or 'id-name' or 'name'
 
 module FindCommon
-  # example: @host = Host.find(params[:id])
   def find_resource
-    not_found and return if params[:id].blank?
```

Comment:
We do, a few lines below we raise `ActiveRecord::NotFound` when the scope is empty - https://github.com/theforeman/foreman/blob/develop/app/controllers/application_controller.rb#L9 , I can also check for params[:id] if needed, but a `not_found` is going to be raised if the resource is not found.